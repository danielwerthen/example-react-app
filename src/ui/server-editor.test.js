import React from 'react';
import ReactDOM from 'react-dom';
import ServerEditor from './server-editor';

const schema = {
  type: "object",
  required: ['name', 'protocol', 'port'],
  properties: {
    name: {
      type: 'string',
    },
    protocol: {
      type: 'string',
      enum: ['HTTP', 'HTTPS'],
    },
    port: {
      type: 'number',
    },
  }
};

describe('ServerEditor', () => {
  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ServerEditor schema={schema} />, div);
    expect(div.innerHTML).toMatchSnapshot();
  });
});
