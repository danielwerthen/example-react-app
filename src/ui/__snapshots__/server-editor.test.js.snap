exports[`ServerEditor should match snapshot 1`] = `"<div data-reactroot=\"\" class=\"server-editor\"><form class=\"rjsf\"><div class=\"form-group field field-object\"><fieldset><div class=\"form-group field field-string\"><label class=\"control-label\" for=\"root_name\">name*</label><div></div><input type=\"text\" id=\"root_name\" label=\"name\" required=\"\" placeholder=\"\" class=\"form-control\" value=\"\"><div></div><div></div></div><div class=\"form-group field field-string\"><label class=\"control-label\" for=\"root_protocol\">protocol*</label><div></div><select id=\"root_protocol\" class=\"form-control\" required=\"\"><option value=\"HTTP\">HTTP</option><option value=\"HTTPS\">HTTPS</option></select><div></div><div></div></div><div class=\"form-group field field-number\"><label class=\"control-label\" for=\"root_port\">port*</label><div></div><input type=\"text\" id=\"root_port\" label=\"port\" required=\"\" placeholder=\"\" class=\"form-control\" value=\"\"><div></div><div></div></div></fieldset><div></div><div></div></div><div class=\"buttons\"><button type=\"submit\">Ok</button><a><button>Cancel</button></a></div></form></div>"`;
